### PR TITLE
Update documenting array arguments in the docstrings of `_qubit_device.py`

### PR DESCRIPTION
**Pull request by antalszava**
Updates one input type and the output type of the `states_to_binary` method.

### CODE REVIEW

File: pennylane/_qubit_device.py [hunk 1]
```diff
@@ -535,14 +535,14 @@ def states_to_binary(samples, num_wires, dtype=np.int64):
         This is an auxiliary method to the generate_samples method.
 
         Args:
-            samples (List[int]): samples of basis states in base 10 representation
+            samples (ndarray): integer samples of basis states in base 10 representation
```
Comment (thisac) [inline pennylane/_qubit_device.py:538]: Not really a change request, but it seems like we're not consistent in using `ndarray`, `np.ndarray` or `numpy.ndarray` at all. 🤔

**Review by thisac — APPROVED** [review]
Thanks @antalszava! 🎉 

Approving this since it's a fairly small, and reasonable change, but with a note that there are several other places (in this file also) that uses `List` when the args/returns are actually `ndarray`s. E.g. in `sample_basis_states` and the probability functions. Not sure if it's worth going through all of them here though.

File: pennylane/_qubit_device.py (hunk 1 above)
Comment (antalszava) [reply pennylane/_qubit_device.py]: Fair point! Was also considering this. We have [`array` mentioned in our guide](https://pennylane.readthedocs.io/en/stable/development/guide/documentation.html#functions-and-methods), changed occurrences in the file to that.

**Comment by antalszava** [comment]
Thanks for the comments @thisac. Made changes in the file itself. Making this change would be out of scope for the entire code base.

File: pennylane/_qubit_device.py [hunk 2]
```diff
@@ -535,14 +535,14 @@ def states_to_binary(samples, num_wires, dtype=np.int64):
...
-            samples (List[int]): samples of basis states in base 10 representation
+            samples (array): integer samples of basis states in base 10 representation
             num_wires (int): the number of qubits
             dtype (type): Type of the internal integer array to be used. Can be
                 important to specify for large systems for memory allocation
                 purposes.
 
         Returns:
-            List[int]: basis states in binary representation
+            array: basis states in binary representation
```
Comment (antalszava) [inline pennylane/_qubit_device.py:545]:
```suggestion
            array[int]: basis states in binary representation
```

File: pennylane/_qubit_device.py [hunk 3]
```diff
@@ -535,14 +535,14 @@ def states_to_binary(samples, num_wires, dtype=np.int64):
         This is an auxiliary method to the generate_samples method.
 
         Args:
-            samples (List[int]): samples of basis states in base 10 representation
+            samples (array): integer samples of basis states in base 10 representation
```
Comment (antalszava) [inline pennylane/_qubit_device.py:538]:
```suggestion
            samples (array[int]): samples of basis states in base 10 representation
```

File: pennylane/_qubit_device.py [hunk 4]
```diff
@@ -516,7 +516,7 @@ def generate_basis_states(num_wires, dtype=np.uint32):
             dtype=np.uint32 (type): the data type of the arrays to use
 
         Returns:
-            np.ndarray: the sampled basis states
+            array: the sampled basis states
```
Comment (antalszava) [inline pennylane/_qubit_device.py:519]:
```suggestion
            array[int]: the sampled basis states
```

**Pull request merged**